go-controller/pkg/ovn/policy_retry.go

```diff
@@ -0,0 +1,186 @@
+package ovn
+
+import (
+	"fmt"
+	knet "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"math/rand"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// iterateRetryNetworkPolicies checks if any outstanding NetworkPolicies exist
+// then tries to re-add them if so
+// updateAll forces all policies to be attempted to be retried regardless
+func (oc *Controller) iterateRetryNetworkPolicies(updateAll bool) {
+	oc.retryNetPolLock.Lock()
+	defer oc.retryNetPolLock.Unlock()
+	now := time.Now()
+	for namespacedName, npEntry := range oc.retryNetPolices {
+		if npEntry.ignore {
+			continue
+		}
+
+		// check if we need to create
+		var policyToCreate *knet.NetworkPolicy
+		if npEntry.newPolicy != nil {
+			// get the latest version of the new policy from the informer, if it doesn't exist we are not going to
+			// create the new policy
+			np, err := oc.watchFactory.GetNetworkPolicy(npEntry.newPolicy.Namespace, npEntry.newPolicy.Name)
+			if err != nil && errors.IsNotFound(err) {
+				klog.Infof("%s policy not found in the informers cache, not going to retry policy create", namespacedName)
+				npEntry.newPolicy = nil
+			} else {
+				policyToCreate = np
+			}
+		}
+
+		npEntry.backoffSec = npEntry.backoffSec * 2
+		if npEntry.backoffSec > 60 {
+			npEntry.backoffSec = 60
+		}
+		backoff := (npEntry.backoffSec * time.Second) + (time.Duration(rand.Intn(500)) * time.Millisecond)
+		npTimer := npEntry.timeStamp.Add(backoff)
+		if updateAll || now.After(npTimer) {
+			klog.Infof("Network Policy Retry: %s retry network policy setup", namespacedName)
+
+			// check if we need to delete anything
+			if npEntry.oldPolicy != nil {
```

Comment:
so essentially this is your trigger right? 

- if oldPolicy is set, you know its a delete retry versus 
- if newpolicy is set (policyToCreate to be more precise), we know its a add retry

if both are set then? you make sure delete is done first and don't retry add huh based on L55's continue?

Reply:
we don't really care what the state is here, we just know if a retry entry has a delete we should remove it first, if that fails we should not try to add the new thing. They may conflict.